### PR TITLE
Student management CRUD - Raveena

### PR DESCRIPTION
**Pull request by RaveenaWickramasinghe**

### CODE REVIEW

File: src/controller/student.controller.js [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+export async function getStudentByID(req, res, next) {
+  if (req.params && req.params.id) {
+    await Student.findById(req.params.id)
+      .populate({
+        path: 'student',
+        populate: {
+          path: 'student',
+          model: 'users',
+          select:
+            '_id firstname lastname dateofbirth address1 address2 city province grade imageurl achievements parent phone email username',
```
Comment (rusiruavb) [inline src/controller/student.controller.js:41]: In enum
'STUDENT_SELECT': '_id firstname lastname dateofbirth address1 address2 city province grade imageurl achievements parent phone email username'

In controller
enums.student.STUDENT_SLECT

**Look question.controller.js file**

**Review by rusiruavb — CHANGES_REQUESTED** [review]
Please make necessary changes

**Review by RaveenaWickramasinghe — COMMENTED** [review]
Changes done

File: src/controller/student.controller.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+import enums from './controller.enums';
+
+export async function createStudent(req, res) {
+  if (req.user && req.user.role === enums.role.ADMIN) {
+    let studentData = new Student(req.body);
+    const student = new Student(studentData);
+    await student
+      .save()
+      .then((data) => {
+        res.status(200).json(data);
```
Comment (rusiruavb) [inline src/controller/student.controller.js:12]: Use response handler

**Review by rusiruavb — CHANGES_REQUESTED** [review]

**Review by RaveenaWickramasinghe — COMMENTED** [review]
Added response handlers to the student controller file

**Review by rusiruavb — APPROVED** [review]
LGTM

**Pull request merged**